includes/TingSearchCqlDoctor.class.inc

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * @file
+ * Class TingSearchCqlDoctor
+ * An attempt to convert non cql search phrases to valid cql.
+ *
+ * examples not valid:
+ * portland (film) => portland and (film)
+ * henning mortensen (f. 1939) => henning and mortensen and "(f. 1939)"
+ * harry and (White night) = harry and "(White night)"
+ *
+ *
+ * valid cql stays the same eg.
+ * dkcclterm.sf=v and dkcclterm.uu=nt and (term.type=bog) not term.literaryForm=fiktion
+ */
+
+class TingSearchCqlDoctor {
+
+  private $cql_string;
+  private $pattern;
+  private $replace;
+  private static $replace_key = 10;
+
+  /**
+   * constructor,
+   * @param $string
+   * sets private member $cql_string with trimmed string
+   */
+  public function __construct($string) {
+    $this->cql_string = trim($string);
+    // handle reserved characters
+    $this->escape_reserved_characters();
+    // remove multiple whitespaces
+    $this->cql_string = preg_replace('/\s+/', ' ', $this->cql_string);
+  }
+
+  /**
+   * Method to convert a string to strict cql (Basically by adding quotes when needed)
+   * @param string $string
+   *   The search query
+   * @return string
+   *   cql compatible string
+   */
+  public function string_to_cql() {
+    // hendle parantheses
+    $this->fix_paranthesis();
+    // handle qoutes
+    $this->fix_qoutes();
+    // format the string
+    $return = $this->format_cql_string();
+    return $return;
+  }
+
+  /**
+   * Format cql string.
+   * @return string; if string is not valid cql it is and'ed
+   */
+  private function format_cql_string(){
+    // last check. All parts of cql string must be valid
```

Comment:
Comments should be complete sentences starting with capital letter and ending with full-stop.